### ISSUE TITLE
readonly files unable to delete attributes

### ISSUE THREAD

**Issue by ksunden** [wright-group/WrightTools#968]
Today in "long tail of relatively easy to fix (but hard to test) consequences of using readonly files in yaqc-cmds":

```
Traceback (most recent call last):
  File "/home/kyle/venvs/pyenv/versions/3.7.6/envs/wright37/lib/python3.7/site-packages/yaqc_cmds/_plot.py", line 126, in on_data_file_written
    self.plot_widget.set_xlim(limits_axis.min(), limits_axis.max())
  File "/home/kyle/venvs/source/WrightTools/WrightTools/_dataset.py", line 395, in min
    min_ = np.nanmin(list(self.chunkwise(f).values()))
  File "/home/kyle/venvs/source/WrightTools/WrightTools/_dataset.py", line 255, in chunkwise
    self._clear_array_attributes_cache()
  File "/home/kyle/venvs/source/WrightTools/WrightTools/_dataset.py", line 109, in _clear_array_attributes_cache
    del self.attrs["max"]
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/home/kyle/venvs/pyenv/versions/3.7.6/envs/wright37/lib/python3.7/site-packages/h5py/_hl/attrs.py", line 105, in __delitem__
    h5a.delete(self._id, self._e(name))
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5a.pyx", line 141, in h5py.h5a.delete
KeyError: 'Unable to delete attribute (no write intent on file)'
```

Basically, `chunkwise` calls the function that clears out cached values of min/max/etc, as chunkwise often changes values of things, and thus invalidates the cache.
This is not always true, though, things like computing the min/max (and indeed anything on a readonly file) will not invalidate the cache.

This is only a problem if one or more of the attributes cached is actually cached and one that is _not_ cached is accessed (or otherwise chunkwise is called)

Operations which change values MUST fail with readonly intent, but will do so anyway.

So, possible fixes:
- do not call `_clear_array_attributes_cache` on readonly files
- try/except in  `_clear_array_attributes_cache` or  `chunkwise`
- add a kwarg to chunkwise to disable the call to `_clear_array_attributes_cache`
- remove the call entirely from chunkwise (good chance it is superfluous, setitem already clears the caches, I think that is the only way we actually change values, at least in chunkwise calls...) (this one makes me slightly nervous, as it is weird behavior of a sequence of events that I doubt is tested, and not sure it would be easy to write such a test...)

In all, none of these proposed fixes are hard to do, just not sure which is the "right" call...

**Comment by ksunden** [comment]
behavior was introduced in #443, dealing with operators.

My belief, though I've only tested with a simple ndarray subclass (rather then our whole wt datasets) is that the implementation _does_ go by way of setitem for our operators (imul, idiv, etc)

```
>>> class Test(np.ndarray):
...   def __setitem__(self, index, value):
...     print("SETITEM CALLED")
...     super().__setitem__(index, value)
...
>>> a = Test(10)
>>> a[:] = 10
SETITEM CALLED
>>> a
Test([10., 10., 10., 10., 10., 10., 10., 10., 10., 10.])
>>> a *=10
>>> a
Test([100., 100., 100., 100., 100., 100., 100., 100., 100., 100.])
>>> a[2] *=10
SETITEM CALLED
>>> a
Test([ 100.,  100., 1000.,  100.,  100.,  100.,  100.,  100.,  100.,
       100.])
>>> a[:] *=10
SETITEM CALLED
```
note that just `a*=10` does _not_ call setitem, but with brackets it does, all of our operators use brackets, due to processing in chunks.

**Comment by ksunden** [comment]
```
>>> d = wt.open(datasets.wt5.v1p0p1_MoS2_TrEE_movie)
>>> d.ai0[0,9,20] = 10
SETITEM CALLED
>>> d.ai0 *= 2
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
SETITEM CALLED
```

Yes indeed, we are calling setitem, many times on chunked datasets

**Comment by ksunden** [comment]
(Also confirmed calling it on unchunked data)

**Comment by ksunden** [comment]
Well, the simple solution (just removing the call in chunkwise) utterly failed in the integration test that is `yaqc-cmds` (not sure why that would be, but got an error about the checksum on the attributes...)

SOOO, one of the more complicated solutions seems necessary

**Comment by ksunden** [comment]
I think we may have fixed this with #983, but also less relevant now that yaqc-cmds plots without wt for live plot.
